modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/transformers/ActionsComponentPropsTransformer.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {render} from 'frontend-js-react-web';
+import React from 'react';
+
+import SidebarPanel from '../SidebarPanel';
+
+export default function propsTransformer({
+	items,
+	sidebarContainerSelector,
+	...otherProps
+}) {
+	const sidebarRefPanel = React.createRef();
+
+	const actions = {
+		showInfo() {
+			showSidebar();
+		},
+		showMetrics() {
+			showSidebar();
+		},
+	};
+
+	const hideSidebar = () => {
+		sidebarRefPanel.current.close();
+	};
+
+	const setSidebarPanelRef = (element) => {
+		sidebarRefPanel.current = element;
+	};
+
+	const showSidebar = () => {
+		if (!sidebarRefPanel.current) {
+			render(
```

Comment:
I don't know if I fully understand the `propsTransformer` attr on the taglib, but it seems weird to be to rendering a separate react tree as a callback for a prop.

I think I am likely missing something here, but maybe you could clarify the `propsTransformer` a bit?

Reply:
It is weird, we discussed it in https://github.com/liferay-frontend/liferay-portal/pull/120

propsTransformer, AFAIK, is nothing more than a layer in between the jsp and the component rendering where we can transform props, for example, to add some listeners like we were doing with the previous `defaultEventHandler` pattern but in a most friendly way for the user.

To me it also seems weir to render anything from here, but for the moment is the easiest way we found to wire this kind of components / tags / jsp / the world 😅